packages/hooks/__tests__/use-intermediate-render.test.ts

```diff
@@ -0,0 +1,98 @@
+import { nextTick, ref } from 'vue'
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { useDelayedRender } from '../use-intermediate-render'
+import type { UseDelayedRenderProps } from '../use-intermediate-render'
+
+describe('useDelayedRender', () => {
+  let props: UseDelayedRenderProps
+  let indicator: any
+  let intermediateIndicator: any
+  let beforeShow: any
+  let afterShow: any
+  let beforeHide: any
+  let afterHide: any
+
+  beforeEach(() => {
+    beforeShow = vi.fn()
+    afterShow = vi.fn()
+    beforeHide = vi.fn()
+    afterHide = vi.fn()
+
+    indicator = ref(false)
+    intermediateIndicator = ref(false)
+
+    props = {
+      indicator,
+      intermediateIndicator,
+      shouldSetIntermediate: vi.fn().mockReturnValue(true),
+      beforeShow,
+      afterShow,
+      beforeHide,
+      afterHide,
+    }
```

Comment:
Perhaps the initialization assignment here can be placed together with the variable declaration. To prevent interference, just reset it in beforeEach.

Reply:
> Perhaps the initialization assignment here can be placed together with the variable declaration. To prevent interference, just reset it in beforeEach.

Sorry, I didn't understand. This code need to reset the variables in beforeEach to reset the mock statistics to zero.

Reply:
Yes, but I think `mockReset` can be called directly to meet the needs.

Reply:
Indeed, `mockRestore` does not reset the number of calls. Maybe we can change our thinking and not reset the function, but match the number of calls.